src/app/file-manager/services/bucket.service.ts

```diff
@@ -2,72 +2,161 @@ import { Injectable } from '@angular/core';
 import { GcsService } from './gcs.service';
 import { Item } from '../models/item';
 import { Observable } from 'rxjs/Observable';
-import { UUID } from 'angular2-uuid';
-import { Type } from '@app/file-manager/models/type';
+import { FirecloudApiService } from '@app/file-manager/services/firecloud-api.service';
+import { SelectionService } from '@app/file-manager/services/selection.service';
 
 @Injectable()
 export class BucketService {
 
   static delimiter: string;
+
+  elements: Item[];
   public isResponseComplete;
-  constructor(private gcsService: GcsService) { }
-  elements: Item[] = [];
+
+  constructor(private gcsService: GcsService,
+    private selectionService: SelectionService) {
+
+  }
   readonly DELIMITER = '/';
 
-  public getBucketData(bucketName: String, delimiter: String, token: String, workspaceName: String): Observable<Item[]> {
-    this.elements = [];
+  public getBucketData(parentNode: Item, token: String, useDelimiter: Boolean, processData: boolean): Observable<Item[]> {
     this.isResponseComplete = false;
-    return this.getBucketFiles(bucketName, token, workspaceName, delimiter);
+    return this.getBucketFiles(parentNode, token, useDelimiter, processData);
```

Comment:
Please add types for all arguments, if possible.

Reply:
??.  I guess this is an error. Types can't be added here.

Reply:
Is there no way to add types to the `getBucketFiles()` method on line 25?

Reply:
ok, I was looking only to line 24. Line 25 is not shown here ... My fault ...